app/src/main/java/swati4star/createpdf/util/TextToPDFUtils.java

```diff
@@ -113,91 +121,69 @@ private BaseColor getBaseColor(int color) {
         );
     }
 
-    /**
-     * Read the .docx file and put it in document
-     *
-     * @param uri      URL to create PDF
-     * @param document PDF Document
-     * @param myfont   Font style in PDF
-     */
-    private void readDocxFile(Uri uri, Document document, Font myfont) {
-        InputStream inputStream;
+    private abstract class FileReader {
```

Comment:
The changes are real good, and align with OOPs concepts.
One suggestion, Can you create different files for `FileReader` and its various implementations? (so that these are not limited to only TextToPDFUtils, and can be used somewhere else also)

Reply:
@Swati4star The FileReaders were separated to different files